lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -267,9 +267,26 @@ static LogicalResult verifyCircuitOp(CircuitOp circuit) {
   // that defines it or, preferentially, the first external module
   // that defines it and has no parameters.
   llvm::DenseMap<Attribute, FExtModuleOp> defnameMap;
+  InnerRefList instanceSyms;
+  InnerRefList leafInnerSyms;
+  SmallVector<NonLocalAnchor> nlaList;
 
   // Verify external modules.
   for (auto &op : *circuit.getBody()) {
+    // Record all the NLAs.
+    if (auto nla = dyn_cast<NonLocalAnchor>(op))
+      nlaList.push_back(nla);
+    else if (auto mod = dyn_cast<FModuleOp>(op)) {
+      auto modName = mod.getNameAttr();
+      mod.walk([&](Operation *op) {
+        if (isa<InstanceOp>(op))
+          instanceSyms.insert(op, modName);
+        else
+          leafInnerSyms.insert(op, modName);
```

Comment:
what is the difference between the sides of the branch? Can we just call `instanceSyms.insert(op, modName);`?